### ISSUE TITLE
using lazy_static on a Vec triggers box_vec

### ISSUE THREAD

**Issue by karyon** [Manishearth/rust-clippy#529]
in servo, this code
```
lazy_static! {
    pub static ref USER_OR_USER_AGENT_STYLESHEETS: Vec<Stylesheet> = {
        let mut stylesheets = vec!();
       ...
        stylesheets
    };
}
```
triggers this warning:
```
<lazy_static macros>:31:22: 31:33 warning: you seem to be trying to use `Box<Vec<T>>`. Consider using just `Vec<T>`, #[warn(box_vec)] on by default
<lazy_static macros>:31 DATA = transmute:: < Box < $ T > , * const $ T > (
                                             ^~~~~~~~~~~
<lazy_static macros>:8:1: 9:72 note: in this expansion of lazy_static! (defined in <lazy_static macros>)
/servo/components/style/selector_matching.rs:54:1: 83:2 note: in this expansion of lazy_static! (defined in <lazy_static macros>)
<lazy_static macros>:31:22: 31:33 help: `Vec<T>` is already on the heap, `Box<Vec<T>>` makes an extra allocation.
for further information visit https://github.com/Manishearth/rust-clippy/wiki#box_vec
```

**Comment by llogiq** [comment]
So either lazy_static is doing something fishy or we need a macro check. Possibly both.

**Comment by Manishearth** [comment]
We need a macro check. I'd prefer one that ensures that the spans of the inner and outer type come from the same syntax context or something.
